### PR TITLE
Python3 changes made and passed the tests with python3

### PR DESCRIPTION
**Pull request by usmansharifkhan**
Python3 changes AP-5226

### CODE REVIEW

File: openidcpy/__init__.py [hunk 1]
```diff
@@ -1 +1 @@
-from client import OidcClient
\ No newline at end of file
+from .client import OidcClient
```
Comment (codemug) [inline openidcpy/__init__.py]: Relative paths are not a great way to resolve dependencies imo. Use `openidcpy.client` instead

File: openidcpy/client.py [hunk 2]
```diff
@@ -91,19 +91,20 @@ def get_tokens_from_code(self, url, redirect_uri, scopes, state):
 
   def validate_jwt(self, token, **kwargs):
     self._discover()
-    if type(token) not in [str, unicode]:
+    if type(token) not in [str, bytes]:
```
Comment (codemug) [inline openidcpy/client.py]: Either a string or bytes should be passed, the handling of both isn't the same. For example, you can't split the token with a `.` below if it's bytes. So ideally, a string should be passed here.

File: openidcpy/__init__.py (hunk 1 above)
Comment (codemug) [inline openidcpy/__init__.py]: Relative paths are not a great way to resolve dependencies imo. Use `openidcpy.client` instead

File: openidcpy/client.py [hunk 3]
```diff
@@ -91,19 +91,20 @@ def get_tokens_from_code(self, url, redirect_uri, scopes, state):
...
     split[0] += "=" * ((4 - len(split[0]) % 4) % 4)
-    key_spec = json.loads(base64.b64decode(split[0]))
+    json_payload = json.loads(base64.b64decode(split[1].encode("utf-8")))
+    key_spec = json.loads(base64.b64decode(split[0].encode("utf-8")))
     key_id = key_spec['kid']
     if key_id not in self.certs:
       raise ValidationError('The token is signed by an unknown key')
     cert = self.certs[key_id]
-    return jwt.decode(token=token, key=cert, **kwargs)
+    return jwt.decode(token=token, key=cert, audience=json_payload['aud'], **kwargs)
```
Comment (codemug) [inline openidcpy/client.py:107]: why do we need to explicitly set the audience?

**Review by codemug — CHANGES_REQUESTED** [review]
Please look into the [str, bytes] issue and the explicit aud property issue

File: openidcpy/client.py (hunk 3 above)
Comment (usmansharifkhan) [reply openidcpy/client.py:107]: In the test 'test_validate_jwt' jwt.decode function is mimicked. This mimicked function takes the audience argument which is then asserted. The test was failing due to audience argument being missing.

**Review by codemug — APPROVED** [review]
LGTM

**Pull request merged**